Hardware/Hardware/CardReader/CardReaderService.swift

```diff
@@ -38,7 +38,9 @@ public protocol CardReaderService {
 
     /// Captures a payment after collecting a payment method succeeds.
     /// The returned publisher will behave as a Future, eventually producing a single value and finishing, or failing.
```

Comment:
I wonder why we need two publishers here instead of just one that is updated with all the events (maybe in an enum?):

- processPaymentCompleted
- cardWasRemoved (or finish in that case)
- errors
- ...
- 😎 

If that is not feasible, could we document that we are now returning a tuple instead of a publisher, and what is the point of each tuple property? Furthermore, as I see it, the future is triggered when the card is removed? I would rename the tuple somehow property to include this meaning. 😄

Reply:
I was really hoping not to make a separate publisher since it just looks so odd and confusing, but I also didn't want to break the existing behavior which is a Future that only triggers completion once as in the original documentation. As you said, the future is only triggered when the card is removed as the last step. However, since we need to log the event in the previous step (client-side processing) we need a separate publisher if we don't want to affect the usage of the main publisher 😓 

I think this tuple of publishers is just too confusing, I'm going to update the PR to allow a future change to run card removal and server-side processing in parallel. Will update!

Reply:
Great! looking forward to seeing that update 👍